File: tests/agent_workloads/adapters/document_markdown_inspection.py
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: document_markdown_inspection INPUT_ROOT")

rows = []
for path in sorted(Path(sys.argv[1]).rglob("*")):
    if not path.is_file():
        continue
    headings = 0
    bullets = 0
    malformed = False
    for line in path.read_text().splitlines():
        stripped = line.strip()
        if line.startswith("#"):
            if stripped == "#":
                malformed = True
            else:
                headings += 1
        if line.startswith("- "):
            bullets += 1
    relative = path.relative_to(sys.argv[1]).as_posix()
    if malformed:
        rows.append(f"reject|{relative}|empty-heading")
    else:
        rows.append(f"document|{relative}|headings={headings}|bullets={bullets}")
for row in sorted(rows):
    print(row)

File: tests/agent_workloads/adapters/document_markdown_inspection.mjs
import { readFileSync, readdirSync, statSync } from "node:fs";
import { join, relative } from "node:path";

if (process.argv.length !== 3) {
  throw new Error("usage: document_markdown_inspection INPUT_ROOT");
}

function files(root) {
  return readdirSync(root, { withFileTypes: true }).flatMap((entry) => {
    const path = join(root, entry.name);
    return entry.isDirectory() ? files(path) : [path];
  });
}

const rows = [];
for (const path of files(process.argv[2]).sort()) {
  if (!statSync(path).isFile()) continue;
  let headings = 0;
  let bullets = 0;
  let malformed = false;
  for (const line of readFileSync(path, "utf8").split(/\r?\n/)) {
    const stripped = line.trim();
    if (line.startsWith("#")) {
      if (stripped === "#") malformed = true;
      else headings += 1;
    }
    if (line.startsWith("- ")) bullets += 1;
  }
  const name = relative(process.argv[2], path).replaceAll("\\", "/");
  if (malformed) {
    rows.push(`reject|${name}|empty-heading`);
  } else {
    rows.push(`document|${name}|headings=${headings}|bullets=${bullets}`);
  }
}
for (const row of rows.sort()) console.log(row);
